### PR TITLE
Add support for Cynthion r0.6, r0.7, r1.0, r1.1, r1.2

### PR DESCRIPTION
**Pull request by mossmann**
The shared USB port (CONTROL) is now handed off to the FPGA at start-up (only on r1 for now). Press the PROGRAM button to have Apollo take over the port.

LED D now indicates that Apollo is using the shared USB port.

### CODE REVIEW

File: firmware/src/boards/luna_d11/usb_switch.c [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ */
+void switch_control_task(void)
+{
+#if ((_BOARD_REVISION_MAJOR_ == 0) && (_BOARD_REVISION_MINOR_ < 6))
+	gpio_set_pin_pull_mode(PROGRAM_BUTTON, GPIO_PULL_UP);
+#else
+	gpio_set_pin_pull_mode(PROGRAM_BUTTON, GPIO_PULL_OFF);
+#endif
+	gpio_set_pin_direction(PROGRAM_BUTTON, GPIO_DIRECTION_IN);
+	gpio_set_pin_direction(FPGA_INT, GPIO_DIRECTION_IN);
```
Comment (mndza) [inline firmware/src/boards/luna_d11/usb_switch.c:52]: `board_init()` in `lib/tinyusb/hw/bsp/samd11/family.c` is configuring another set of GPIO pins, including `BUTTON_PIN` (`PA16` regardless of the board revision).

I think we should either move the GPIO configuration lines to `board_init()` or create a new one somewhere else, and only keep the `gpio_get_pin_level()` calls here.

**Review by Vangog7770 — CHANGES_REQUESTED** [review]
- [ ] 1. ## ### **__**

File: firmware/src/boards/cynthion_d11/usb_switch.c [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+ * Handle switch control user request.
+ */
+void switch_control_task(void)
+{
+#if ((_BOARD_REVISION_MAJOR_ == 0) && (_BOARD_REVISION_MINOR_ < 6))
+	gpio_set_pin_pull_mode(PROGRAM_BUTTON, GPIO_PULL_UP);
+#else
+	gpio_set_pin_pull_mode(PROGRAM_BUTTON, GPIO_PULL_OFF);
+#endif
+	gpio_set_pin_direction(PROGRAM_BUTTON, GPIO_DIRECTION_IN);
```
Comment (mndza) [inline firmware/src/boards/cynthion_d11/usb_switch.c:51]: This is no longer necessary, the GPIO is configured here:
https://github.com/hathach/tinyusb/blob/5b08a65ad2e695332d56bd4676841a91f34dd5b1/hw/bsp/samd11/family.c#L80-L81

**Review by mndza — APPROVED** [review]
:+1:

**Pull request merged**